src/pip/_internal/commands/uninstall.py

```diff
@@ -47,6 +51,13 @@ def add_options(self) -> None:
                 "file.  This option can be used multiple times."
             ),
         )
+        self.cmd_opts.add_option(
+            "-a",
+            "--all",
+            dest="all",
+            action="store_true",
+            help="Uninstall all Installed packages.",
+        )
```

Comment:
We should make this mutually exclusive to `-r` and names passed in from arguments, so a user wouldn’t accidentally destroy an environment with e.g. `pip install -a -r r.txt` and `pip install -a mypackage`

Reply:
I think I could not understand what you meant. Can you please elaborate?

Reply:
This is an option on `pip uninstall` and doesn't affect `pip install`.